src/pages/index.js

```diff
@@ -54,34 +94,97 @@ const formValidatorProfile = new FormValidator(
 );
 formValidatorProfile.enableValidation();
 
-//render initial cards
 function handleImageClick(data) {
   imagePreviewPopup.open(data);
 }
+function handleDeleteButtonClick(id, card) {
+  DeleteConfirmationPopup.open();
+  DeleteConfirmationPopup.setSubmitAction(() => {
+    api
+      .deleteCard(id)
+      .then(() => {
+        DeleteConfirmationPopup.close();
+        card.deleteCardFromPage();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleLikeButtonClick(id, card, isLiked) {
+  if (isLiked) {
+    api
+      .removeLike(id)
+      .then((res) => {
+        card.toggleLikeOnPage(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .addLike(id)
+      .then((res) => {
+        card.toggleLikeOnPage(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
 function createCard(item) {
-  const cardElement = new Card(item, cardSelector, () =>
-    handleImageClick(item)
+  const cardElement = new Card(
+    item,
+    cardSelector,
+    () => handleImageClick(item),
+    handleDeleteButtonClick,
+    handleLikeButtonClick
   );
   return cardElement.generateCard();
 }
-function renderCard(item) {
-  const cardElement = createCard(item);
-  cardsList.addItem(cardElement);
-}
-const cardsList = new Section(
-  { items: initialCards, renderer: renderCard },
-  cardListSectionSelector
-);
-cardsList.renderItems();
 
-//popup with form for adding a place
+api
+  .getInitialCards()
+  .then((result) => {
+    const cardsList = new Section(
+      { items: result, renderer: createCard },
+      cardListSectionSelector
+    );
+    cardsList.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);`